HeadlessChrome: modify config via event listener

limenet opened the pull request:
## Changes in this pull request  

When generating a PDF based on a Document in `\Pimcore\Web2Print\Processor::startPdfGeneration`, a `pimcore.document.print.prePdfGeneration` is dispatched (line 125). The payload (a `\Pimcore\Event\Model\DocumentEvent`) is then passed (line 127) as `$config` to `\Pimcore\Web2Print\Processor::buildPdf`.

Currently, the implementation of `buildPdf()` in `\Pimcore\Web2Print\Processor\HeadlessChrome` does not use the argument `$config`, rendering any changes to the job config in the event listener useless. This PR addresses this issue by merging the global Web2Print options with the options passed in that event.

## Additional info  

Since there still is [a 10.5.x milestone](https://github.com/pimcore/pimcore/milestone/183) and the change is a bugfix, this PR targets `10.5`.

A sample listener might look as follows:

```php
<?php

namespace App\EventListener\Document;

use Pimcore\Event\Model\DocumentEvent;

class ConfigListener
{
    public function setParams(DocumentEvent $event): void
    {
        $jobConfig = $event->getArgument('jobConfig');

        $jobConfig->headlessChromeSettings = [
            'marginTop' => '0mm',
            'marginRight' => '0mm',
            'marginBottom' => '0mm',
            'marginLeft' => '0mm',
        ];

        $event->setArgument('jobConfig', $jobConfig);
    }
}
```

limenet committed:
merge global options with options passed via event

dvesh3 commented on lib/Web2Print/Processor/HeadlessChrome.php:
```diff
@@ -38,6 +38,7 @@ protected function buildPdf(Document\PrintAbstract $document, $config)
         $web2printConfig = Config::getWeb2PrintConfig();
         $web2printConfig = $web2printConfig->get('headlessChromeSettings');
         $web2printConfig = json_decode($web2printConfig, true);
+        $web2printConfig = array_merge($web2printConfig, ($config->headlessChromeSettings ?? []));
```
could you please also update arguments [here ](https://github.com/pimcore/pimcore/blob/723d46a2be247f81e4776cec90b104796dbe30c6/lib/Event/DocumentEvents.php#L121)and also add headless chrome example on how to extend headless chrome settings in the comments? thanks!

limenet committed:
use modifyConfig event to modify HeadlessChrome options

limenet commented:
While working on your suggested changes -- thanks! -- I realized my changes don't follow the pattern in the other processors where the modifications in question instead use `\Pimcore\Event\DocumentEvents::PRINT_MODIFY_PROCESSING_CONFIG`.

(`\Pimcore\Event\DocumentEvents::PRINT_PRE_PDF_GENERATION` is used in conjunction with `\Pimcore\Web2Print\Processor::getProcessingOptions` which is used for the options exposed in the Pimcore Admin.)

I refactored my code accordingly and moved the handling from `\Pimcore\Web2Print\Processor\HeadlessChrome::buildPdf` to `\Pimcore\Web2Print\Processor\HeadlessChrome::getPdfFromString` similar to the handling in `\Pimcore\Web2Print\Processor\WkHtmlToPdf`.

The updated example looks as follows:

```php
<?php

namespace App\EventListener\Document;

use Pimcore\Event\Model\PrintConfigEvent;

class ConfigListener
{
    public function setParams(PrintConfigEvent $event): void
    {
        $params = [
            'margin' => [
                'top' => '0mm',
                'bottom' => '0mm',
                'right' => '0mm',
                'left' => '0mm',
            ],
        ];

        $event->setArgument('params', $params);
    }
}
```

dvesh3 commented:
Now, this even works well with default options when no `web-to-print` config is there. thanks for the changes 👍

limenet commented:
You're welcome, thanks for reviewing!
